master/pkg/model/environment_config.go

```diff
@@ -38,7 +38,10 @@ type Environment struct {
 
 // RuntimeItem configures the runtime image.
 type RuntimeItem struct {
-	CPU string `json:"cpu,omitempty"`
+	CPU  string `json:"cpu,omitempty"`
+	CUDA string `json:"cuda,omitempty"`
+	ROCM string `json:"rocm,omitempty"`
+	// Deprecated: use CUDA instead.
 	GPU string `json:"gpu,omitempty"`
```

Comment:
No need to have deprecated fields in go structs, since there's no user code with access to these.  Structuring it this way just leaks deprecated configs into the wider go code.